packages/pharos/src/components/toggle-button-group/pharos-toggle-button-group.ts

```diff
@@ -0,0 +1,164 @@
+
+  protected firstUpdated(): void {
+    this.addEventListener('pharos-toggle-button-selected', this._handleButtonSelected);
+    this.addEventListener('keydown', this._handleKeydown);
+    this.addEventListener('focusout', this._handleFocusout);
+
+    const toggleButtons = this.querySelectorAll(
+      `pharos-toggle-button`
+    ) as NodeListOf<PharosToggleButton>;
+    this._selectInitialToggleButton(toggleButtons);
+
+    const maxIdx = toggleButtons.length - 1;
+    toggleButtons.forEach((button, idx) => {
+      button['_first'] = idx == 0;
+      button['_last'] = idx == maxIdx;
+    });
+  }
+
+  private _selectInitialToggleButton(toggleButtons: NodeListOf<PharosToggleButton>): void {
+    const selected = this.querySelector(`pharos-toggle-button[selected]`) as PharosToggleButton;
+    const selectedButton: PharosToggleButton = selected ? selected : toggleButtons[0];
+
+    selectedButton.selected = true;
+  }
+
+  private _handleButtonSelected(event: Event): void {
+    const selected = event.target as PharosToggleButton;
+
+    const previous = this.querySelector(
+      `pharos-toggle-button[selected]:not([id="${selected.id}"])`
+    ) as PharosToggleButton;
+
+    if (previous) {
+      previous.selected = false;
+    }
+
+    const toggleButtons = Array.prototype.slice.call(
+      this.querySelectorAll(`pharos-toggle-button`)
+    ) as PharosToggleButton[];
+    let selectIdx = -1;
+    toggleButtons.forEach((b, idx) => {
+      b['_hideLeftBorder'] = false;
+      b['_hideRightBorder'] = false;
+      if (b.id == selected.id) {
+        selectIdx = idx;
+      }
+    });
+
+    if (selectIdx > 0) {
+      toggleButtons[selectIdx - 1]['_hideRightBorder'] = true;
+    }
+    if (selectIdx < toggleButtons.length - 1) {
+      toggleButtons[selectIdx + 1]['_hideLeftBorder'] = true;
+    }
+  }
+
+  private _handleKeydown(event: KeyboardEvent): void {
+    switch (event.key) {
+      case 'Right':
+      case 'ArrowRight':
+        event.preventDefault();
+        this._handleArrowKeys(true);
+        break;
+      case 'Left':
+      case 'ArrowLeft':
+        event.preventDefault();
+        this._handleArrowKeys(false);
+        break;
+      case 'Enter':
+      case ' ':
+      case 'Spacebar':
+        event.preventDefault();
+        this._handleEnterKey();
+        break;
+    }
+  }
+
+  private async _handleArrowKeys(moveForward: boolean): Promise<void> {
+    const toggleButtons = Array.prototype.slice.call(
+      this.querySelectorAll(`pharos-toggle-button`)
+    ) as PharosToggleButton[];
+    const ids = toggleButtons.map((toggleButton) => toggleButton.id);
+
+    const focused = document.activeElement as PharosToggleButton;
+    if (!focused[matchesFunc]('pharos-toggle-button')) {
+      return;
+    }
+
+    let index = ids.findIndex((v) => v === focused.id);
+
+    if (moveForward) {
+      index = index === ids.length - 1 ? 0 : index + 1;
+    } else {
+      index = index === 0 ? ids.length - 1 : index - 1;
+    }
```

Comment:
Having a little bit of a hard time wrapping my head around this block. Seems like we are trying to get either the next or previous `toggleButton` in the array? Wondering if there is an opportunity to extract this functionality out into it's own function. It feels like we are doing something similar in `_handleButtonSelected` too?

Reply:
This code actually works a little differently than the previous code. If you're using the right arrow key, it moves forward until it hits the last button, when it goes back to the beginning. Similarly, if you're using the left arrow key, it moves backward until it hits the first button.

I just rewrote the code in `_handleButtonSelected` such that I don't think we can reasonably combine them but I will try to rewrite this to make it a bit clearer.

Reply:
Gotcha, I think I am following more. I am wondering if we need the `ids` array we are mapping. Could we get away with just doing:

`let focusedToggleButtonIndex = toggleButtons.findIndex((toggleButton) => toggleButton.id === focused.id)`

Reply:
fixed